drivers/soundwire/slave.c

```diff
@@ -39,12 +39,12 @@ int sdw_slave_add(struct sdw_bus *bus,
 
 	if (id->unique_id == SDW_IGNORED_UNIQUE_ID) {
 		/* name shall be sdw:link:mfg:part:class */
-		dev_set_name(&slave->dev, "sdw:%x:%x:%x:%x",
+		dev_set_name(&slave->dev, "sdw:%x:%04x:%04x:%02x",
 			     bus->link_id, id->mfg_id, id->part_id,
 			     id->class_id);
 	} else {
 		/* name shall be sdw:link:mfg:part:class:unique */
-		dev_set_name(&slave->dev, "sdw:%x:%x:%x:%x:%x",
+		dev_set_name(&slave->dev, "sdw:%x:%04x:%04x:%02x:%01x",
```

Comment:
why we don't set %01x for link_id ? link_id may be bigger than 9 ?  thanks!

Reply:
I'll leave that to your Soundwire folks to decide. :) Probably a larger-than-9 link-id is very rare and having a variable number at the start won't hurt readability as much.

Reply:
Good comment. I thought the link ID was not bound but in the MIPI DisCo spec it uses bits 51:48, so it should be %01x for consistency. Will fix.